pkg/mod/gomod.go

```diff
@@ -22,23 +21,26 @@ type goModule struct {
 }
 
 func goGetByFilepath(filename string) error {
-	dir := filepath.Dir(filename)
-	re := regexp.MustCompile(`^\.|\.\.|/$`)
+	names := strings.Split(filename, "/")
+	if len(names) > 0 {
+		gogetPath := names[0]
+
+		for i := 1; i < len(names); i++ {
+			err := goGet(gogetPath)
+			if err == nil {
+				return nil
+			}
+			logrus.Debugf("go get %s error: %s\n", gogetPath, err.Error())
 
-	for !re.MatchString(dir) {
-		err := goGet(dir)
-		if err == nil {
-			return nil
+			gogetPath = path.Join(gogetPath, names[i])
```

Comment:
Alternatively, slightly less efficient, but cleaner (And who cares about efficiency when you're hitting the filesystem? :smile:):
```go
for i = range names[1:] {
	err := goGet(path.Join(names[:1+i]...))
	if err == nil {
		return nil
	}
	logrus.Debugf("go get %s error: %s\n", gogetPath, err.Error())
}
```